spectator_mode/init.lua

```diff
@@ -127,8 +127,7 @@ end
 
 minetest.register_on_joinplayer(function(player)
 	minetest.after(3, function(name)
-		local player = minetest.get_player_by_name(name)
-		if not player then
+		if not minetest.get_player_by_name(name) then
```

Comment:
Whoa, why has this been removed? It's inefficient to call `minetest.get_player_by_name` every time we need to access the player ObjectRef. That's why we had locally cached the player object first (`local player`) and used it for the subsequent calls below.

Also, removing this would break the rest of this callback, btw.